src/smt/smt_engine_check_proof.cpp

```diff
@@ -15,6 +15,12 @@
  ** \todo document this file
  **/
 
+#if defined(_MSC_VER) || defined(__MINGW32__) || defined(__MINGW64__)
```

Comment:
This isn't to fix compilation issues; it's to fix proof checking on Windows. The issue with the original code is that `mkstemp` opens its temporary file without the modes required for shared access, which causes the attempt to open a second handle (the `std::ofstream` on line 112) to fail. Since that fails, the temporary file never gets the proof output written to it, which causes most of the test suite to fail, since the temporary file is empty every time.